ansible/roles/sshkey-host-access/tasks/main.yml

```diff
@@ -27,35 +27,33 @@
         owner: "{{ KEY_LOC_OWNER }}"
         group: "{{ KEY_LOC_GROUP }}"
 
+    - name: Ensure correct ownership of public key
+      file:
+        path: "{{ FULL_PUBKEY_PATH }}"
+        owner: "{{ KEY_LOC_OWNER }}"
+        group: "{{ KEY_LOC_GROUP }}"
+
     - name: add the key name to .default_ids
       lineinfile:
-        dest: "{{ KEY_LOC_PATH }}/.default_ids"
-        line: "{{ ATMOUSERNAME }}{{ KEY_LOC_PREFIX }}_default"
+        dest: "{{ KEY_LOCAL_DIR }}/.default_ids"
+        line: "{{ USERNAME }}{{ KEY_LOC_PREFIX }}_default"
         create: yes
         state: present
         owner: "{{ KEY_LOC_OWNER }}"
         group: "{{ KEY_LOC_GROUP }}"
 
-    - name: finally, transfer the public key locally
+    - name: transfer the public key from {{ EXTERNAL_HOST }} host to controller
       fetch:
         src: "{{ FULL_PUBKEY_PATH }}"
         dest: "{{ LOCAL_PUBKEY_DIR }}"
```

Comment:
Since the variable was changed, this should be `dest: "{{ LOCAL_PUBKEY_DIR }}/"`  (weird behavior of `fetch` module). Without it, the fetched file on the local system would be located at `{{ LOCAL_PUBKEY_DIR }}/<hostname>/{{ FULL_PUBKEY_PATH }}` which uses the full path of the pubkey and embeds the hostname in there, rather than just the basename. Weird stuff.